features/publish.feature

```diff
@@ -106,7 +106,7 @@ Feature: Publish reports
   @spawn
   Scenario: when results are not published due to an error raised by the server, the banner is displayed
     When I run cucumber-js with env `CUCUMBER_PUBLISH_TOKEN=keyboardcat`
-    Then it fails
+    Then it passes
```

Comment:
The only functional change here. For formatters, we catch errors on the stream(s), log them and then fail the test run at the end. Publish is now not a formatter, and at any rate this felt a bit off - failing to publish the report seems like it shouldn't fail the test run? That's debatable, though.

It does raise a couple of questions about error handling though:
- If there's an uncaught error in a plugin, should we try to fail the test run gracefully-ish?
- Should there be a way for a plugin to say "The test run should be marked as a failure because XYZ"? Or should it just throw if it wants to do that?